### PR TITLE
bpo-36250: ignore ValueError from signal in non-main thread

### PR DESCRIPTION
**Pull request by blueyed**
https://bugs.python.org/issue36250

### CODE REVIEW

File: Lib/pdb.py [hunk 1]
```diff
@@ -341,8 +341,12 @@ def preloop(self):
     def interaction(self, frame, traceback):
         # Restore the previous signal handler at the Pdb prompt.
         if Pdb._previous_sigint_handler:
-            signal.signal(signal.SIGINT, Pdb._previous_sigint_handler)
-            Pdb._previous_sigint_handler = None
+            try:
+                signal.signal(signal.SIGINT, pdb.Pdb._previous_sigint_handler)
```
Comment (zware) [inline Lib/pdb.py]:
```suggestion
                signal.signal(signal.SIGINT, Pdb._previous_sigint_handler)
```

File: Lib/pdb.py [hunk 2]
```diff
@@ -341,8 +341,12 @@ def preloop(self):
...
+            except ValueError:  # ValueError: signal only works in main thread
+                pass
+            else:
+                pdb.Pdb._previous_sigint_handler = None
```
Comment (zware) [inline Lib/pdb.py]:
```suggestion
                Pdb._previous_sigint_handler = None
```

**Review by zware — CHANGES_REQUESTED** [review]

**Comment by blueyed** [comment]
I have made the requested changes; please review again.

**Review by zware — APPROVED** [review]

**Comment by zware** [comment]
Thanks for the patch, and sorry for the delay in my re-review!

**Pull request merged**